src/backend/distributed/deparser/citus_ruleutils.c

```diff
@@ -1381,6 +1381,13 @@ contain_nextval_expression_walker(Node *node, void *context)
 			return true;
 		}
 	}
+
+	if (IsA(node, Query))
```

Comment:
I think we should try to avoid using `query_tree_walker` inside `_expression_walker`.

In general, those two concepts are use separately in PG/Citus code bases.

For query, can't we use `FindNodeMatchingCheckFunction((Node *) query, IsNextValExpr);`

Where `IsNextValExpr`

```C
bool
IsNextValExpr(Node *node)
{
	/* check if the node contains an identity column */
	if (IsA(node, NextValueExpr))
	{
		return true;
	}

	/* check if the node contains call to 'nextval' */
	if (IsA(node, FuncExpr))
	{
		FuncExpr *funcExpr = (FuncExpr *) node;

		if (funcExpr->funcid == F_NEXTVAL)
		{
			return true;
		}
	}
	return false;
}
```

Reply:
>>I think we should try to avoid using query_tree_walker inside _expression_walker.

OK, I followed the recommendation from the PG header comments of the `expression_tree_walker`
![image](https://github.com/citusdata/citus/assets/44680808/1440e118-c614-4418-a3ef-a43a8ddd698e)

Let me try this option

Reply:
Made changes using `FindNodeMatchingCheckFunction`